imap/jmap_api.c

```diff
@@ -1018,8 +1018,15 @@ HIDDEN void jmap_accounts(json_t *accounts, json_t *primary_accounts)
 {
     /* Find shared accounts */
     strarray_t patterns = STRARRAY_INITIALIZER;
-    char *userpat = xstrdup("user.*");
-    userpat[4] = jmap_namespace.hier_sep;
+    char *userpat = NULL;
+    if (jmap_namespace.isalt) {
+        userpat = xstrdup("Other Users.*");
+        userpat[11] = jmap_namespace.hier_sep;
+    }
```

Comment:
The "Other Users" string comes from the imapd.conf `userprefix` setting, so this should use `config_getstring(IMAPOPT_USERPREFIX)` rather than hard-coding "Other Users".

You might need to use a `struct buf` to assemble the parts, since if you just xstrdup it like the current code it'll be missing the ".*" at the end... maybe something like (untested):

```
if (jmap_namespace.isalt) {
    struct buf buf = BUF_INITIALIZER;
    buf_printf(&buf, "%s%c*", config_getstring(IMAPOPT_USERPREFIX), jmap_namespace.hier_sep);
    userpat = buf_release(&buf);
}
```

That buf_printf line probably needs to be wrapped, but I'm typing in a github comment box, not a code editor 😅

Reply:
I'd like use `xstrndup` to avoid buf_release operation, and I must compute strlen(config_getstring(IMAPOPT_USERPREFIX)) 
 here to replace `jmap_namespace.hier_sep`.

Reply:
After comparing their implementation, I'll change `xstrndup` to `struct buf`.